bb8/src/api.rs

```diff
@@ -406,8 +408,12 @@ pub trait CustomizeConnection<C: Send + 'static, E: 'static>:
     ///
     /// The default implementation simply returns `Ok(())`. If this method returns an
     /// error, it will be forwarded to the configured error sink.
-    async fn on_acquire(&self, _connection: &mut C) -> Result<(), E> {
-        Ok(())
+    fn on_acquire<'a, 'b, 'r>(&'a self, _connection: &'b mut C) -> BoxFuture<'r, Result<(), E>>
```

Comment:
Can we spell this `<'r, 'a: 'r, 'b: 'r>` instead? And do we really need the 3 separate lifetimes?

Why is `BoxFuture` useful? Could we just do `Box<dyn Future<..>>` instead?

Reply:
`BoxFuture<'a, T>` is actually `Pin<Box<dyn Future<Output = T> + Send + 'a>>`. I'll apply the suggestions.

Reply:
> And do we really need the 3 separate lifetimes?
I'm not sure, but that's what `async_trait` generated, so I kept it to prevent further breaking changes in the API.